protocol-specs/generators/multisignature_registration/index.js

```diff
@@ -14,164 +14,206 @@
+		privateKey: Buffer.from(
 			'ddc8e19d6697d6e5c1dacf6576a7169752810999918212afe14d3978b354f8aa4a67646a446313db964c39370359845c52fce9225a3929770ef41448c258fd39',
-		publicKey:
+			'hex',
+		),
+		publicKey: Buffer.from(
 			'4a67646a446313db964c39370359845c52fce9225a3929770ef41448c258fd39',
-		address: '3372320078773139180L',
+			'hex',
+		),
+		address: Buffer.from('ecb6308c3ee3cc2ed1fa266b85ba127d63a4ee1c', 'hex'),
 	},
 	optionalOne: {
 		passphrase:
 			'sugar object slender confirm clock peanut auto spice carbon knife increase estate',
-		privateKey:
+		privateKey: Buffer.from(
 			'69aa94ea7ade3b7b08e277b18c1a590b2306ce5973ae8462b0b85122b180e89c57df5c3811961939f8dcfa858c6eaefebfaa4de942f7e703bf88127e0ee9cca4',
-		publicKey:
+			'hex',
+		),
+		publicKey: Buffer.from(
 			'57df5c3811961939f8dcfa858c6eaefebfaa4de942f7e703bf88127e0ee9cca4',
-		address: '7745870967079479156L',
+			'hex',
+		),
+		address: Buffer.from('74a7c8ec9adc7e6ba5c1cf9410d5c6c6bf6aba7d', 'hex'),
 	},
 	optionalTwo: {
 		passphrase:
 			'faculty inspire crouch quit sorry vague hard ski scrap jaguar garment limb',
-		privateKey:
+		privateKey: Buffer.from(
 			'ffed38380998a90a2af9501f10182bc2a07922448ab383575b1e34aeddfa5482fa406b6952d377f0278920e3eb8da919e4cf5c68b02eeba5d8b3334fdc0369b6',
-		publicKey:
+			'hex',
+		),
+		publicKey: Buffer.from(
 			'fa406b6952d377f0278920e3eb8da919e4cf5c68b02eeba5d8b3334fdc0369b6',
-		address: '7086965981385941478L',
+			'hex',
+		),
+		address: Buffer.from('e661c9ff02f65962ac08bc79a2f5c0d44b312fbc', 'hex'),
 	},
 };
 
-const SIGNATURE_NOT_PRESENT = Buffer.from('00', 'hex');
-const SIGNATURE_PRESENT = Buffer.from('01', 'hex');
+const outputBase64Account = account => ({
+	...account,
+	privateKey: account.privateKey.toString('base64'),
+	publicKey: account.publicKey.toString('base64'),
+	address: account.address.toString('base64'),
+});
 
-const sortKeysAscending = publicKeys =>
-	publicKeys.sort((publicKeyA, publicKeyB) => {
-		// eslint-disable-next-line no-undef, new-cap
-		if (BigInt(`0x${publicKeyA}`) > BigInt(`0x${publicKeyB}`)) {
-			return 1;
-		}
-		// eslint-disable-next-line no-undef, new-cap
-		if (BigInt(`0x${publicKeyA}`) < BigInt(`0x${publicKeyB}`)) {
-			return -1;
-		}
-		return 0;
-	});
-
-const getId = transactionBytes => {
-	const transactionHash = hash(transactionBytes);
-	const transactionId = bufferToHex(transactionHash);
-
-	return transactionId;
+const baseSchema = {
+	$id: 'baseSchema',
+	type: 'object',
+	required: ['type', 'nonce', 'fee', 'senderPublicKey', 'asset'],
+	properties: {
+		type: {
+			dataType: 'uint32',
+			fieldNumber: 1,
+		},
+		nonce: {
+			dataType: 'uint64',
+			fieldNumber: 2,
+		},
+		fee: {
+			dataType: 'uint64',
+			fieldNumber: 3,
+		},
+		senderPublicKey: {
+			dataType: 'bytes',
+			fieldNumber: 4,
+		},
+		asset: {
+			dataType: 'bytes',
+			fieldNumber: 5,
+		},
+		signatures: {
+			type: 'array',
+			items: {
+				dataType: 'bytes',
+			},
+			fieldNumber: 6,
+		},
+	},
```

Comment:
Same here, this base schema will be used among many places, can we move it to a shared source file?